src/modules/commander/Arming/PreFlightCheck/checks/ekf2Check.cpp

```diff
@@ -79,16 +79,24 @@ bool PreFlightCheck::ekf2Check(orb_advert_t *mavlink_log_pub, vehicle_status_s &
 
 			} else if (status.pre_flt_fail_innov_height) {
 				mavlink_log_critical(mavlink_log_pub, "Preflight Fail: height estimate not stable");
-
-			} else if (status.pre_flt_fail_mag_field_disturbed) {
-				mavlink_log_critical(mavlink_log_pub, "Preflight Fail: strong magnetic interference detected");
 			}
 		}
 
 		success = false;
 		goto out;
 	}
 
+	param_get(param_find("COM_ARM_MAG_STR"), &is_mag_strength_check);
```

Comment:
I would fill the local variable when it's declared. Ideally here but if you have to declare it earlier because of the **** gotos I would move the `param_get()` there as well.

Reply:
I hope we can get rid of those ugly goto at some point...